pandas/core/resample.py

```diff
@@ -1622,19 +1673,30 @@ def _get_timestamp_range_edges(first, last, offset, closed="left", base=0):
         The beginning Timestamp of the range to be adjusted.
     last : pd.Timestamp
         The ending Timestamp of the range to be adjusted.
-    offset : pd.DateOffset
+    freq : pd.DateOffset
         The dateoffset to which the Timestamps will be adjusted.
     closed : {'right', 'left'}, default None
         Which side of bin interval is closed.
-    base : int, default 0
-        The "origin" of the adjusted Timestamps.
+    origin : pd.Timestamp, default None
+        The timestamp on which to adjust the grouping. It must be timezone
+        aware if the index of the resampled data is. If None is passed, the
+        first day of the time series at midnight is used.
+    offset : pd.Timedelta, default is None
+        An offset timedelta added to the origin.
 
     Returns
     -------
     A tuple of length 2, containing the adjusted pd.Timestamp objects.
     """
-    if isinstance(offset, Tick):
-        if isinstance(offset, Day):
+    if isinstance(freq, Tick):
+        is_idx_tz_aware = first.tzinfo is not None or last.tzinfo is not None
+        if origin is not None and origin.tzinfo is None and is_idx_tz_aware:
+            raise ValueError(
+                "The origin must be timezone aware when the index "
```

Comment:
`"The origin must have the same timezone as the index."`

Reply:
I did the change. However, I am not sure this entirely true since the `origin` just needs to be timezone aware when the index is. But I agree with you since it is less confusing for the reader anyway.